sotodlib/io/load_smurf.py

```diff
@@ -896,169 +899,43 @@ def load_data(self, start, end, stream_id=None, detset=None, show_pb=True, load_
+        if stream_id is not None:
+            q.filter(Files.stream_id == stream_id)
+
+        q = q.order_by(Files.start).distinct()
+        flist = [x[0] for x in q.all()]
 
         if detset is None:
             detset = session.query(Detsets).filter(Detsets.start <= start)
             detset = detset.order_by( db.desc(Detsets.start) ).first() 
         else:
             detset = session.query(Detsets).filter(Detsets.name==detset).one()
-
-        samples, channels = 0, 0
-        num_frames = 0
-        for f in frames:
-            num_frames += 1
-            samples += f.n_samples
-            channels = max(f.n_channels, channels)
-
-        timestamps = np.full((samples,), np.nan, dtype=np.float64)
-        data = np.full((channels, samples), 0, dtype=np.int32)
-        if load_biases:
-            biases = np.full((num_bias_lines, samples), 0, dtype=np.int32)
-        else:
-            biases = None
-
-        primary = {}
-
-        cur_sample = 0
-        cur_file = None
-        timing_paradigm = None
-        for frame_info in tqdm(frames, total=num_frames, disable=(not show_pb)):
-            file = frame_info.file.path
-            if file != cur_file:
-                reader = so3g.G3IndexedReader(file)
-                cur_file = file
-
-            reader.Seek(frame_info.offset)
-            frame = reader.Process(None)[0]
-            nsamp = frame['data'].n_samples
-
-            key_order = [int(k[1:]) for k in frame['data'].keys()]
-            data[key_order, cur_sample:cur_sample + nsamp] = frame['data']
-
-            if load_biases:
-                bias_order = [int(k[-2:]) for k in frame['tes_biases'].keys()]
-                biases[bias_order, cur_sample:cur_sample + nsamp] = frame['tes_biases']
-
-            # Loads primary data
-            if 'primary' in frame.keys():
-                for k, v in frame['primary'].items():
-                    if k not in primary:
-                        primary[k] = np.zeros(samples, dtype=np.int64)
-                    primary[k][cur_sample:cur_sample + nsamp] = v
-
-            ts, paradigm = get_sample_timestamps(frame)
-            if timing_paradigm is None:
-                timing_paradigm = paradigm
-            elif timing_paradigm != paradigm:
-                timing_paradigm = TimingParadigm.Mixed
-
-            timestamps[cur_sample:cur_sample + nsamp] = ts
-
-            cur_sample += nsamp
-
-        # Conversion from DAC counts to squid phase
-        rad_per_count = np.pi / 2**15
-        data = data * rad_per_count
-
-        if len(timestamps) > 0:
-            status = self.load_status(timestamps[0])
-        else:
+        
+        scan_start = session.query(Frames.start).filter(Frames.start > start,
+                                                        Frames.type_name=='Scan')
+        scan_start = scan_start.order_by(Frames.start).first()
+        
+        try:
+            status = self.load_status(scan_start[0])
```

Comment:
One thing I noticed when trying to debug why some queries failed to load the status, is that it's actually possible for the first timestamp of a scan frame to occur *before* the first Status frame.  This is actually how I intended the Smurf SessionManager to work, but in some cases it results in an Observation, Wiring, and Scan frame all to be written at time `t`, but the first data sample actually being at `t-1`. 

tldr: we should be loading the Status frame at something like `Frame.time+1` to be safe.